packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -308,6 +328,30 @@ class TFContracts extends Contracts {
     return res;
   }
 
+  /**
+   * Calculates the overdue amount for a contract.
+   *
+   * This method calculates the overdue amount by summing the overdraft amounts
+   * from the contract payment state and multiplying it by the product of:
+   * 1. The time elapsed since the last billing in seconds (with an additional time allowance).
+   * 2. The contract cost per second.
+   *
+   * The resulting overdue amount represents the amount that needs to be addressed.
+   *
+   * @param {CalculateOverdueOptions} options - The options containing the contract ID.
+   * @returns {Promise<number>} - The calculated overdue amount.
+   */
+  async calculateContractOverDue(options: CalculateOverdueOptions) {
+    const { lastUpdatedSeconds, standardOverdraft, additionalOverdraft } = options.paymentState;
+    const contractCost = await this.getConsumption({ id: options.id, graphqlURL: options.graphqlURL });
```

Comment:
This is just an estimate, as some contracts might have public IP addresses, and the cost will depend on the amount of bandwidth used, which is likely to vary.

The cost calculation should simulate the billing process on tfchain to provide a more accurate estimate. This involves checking the pricing policy, NU, and other resource consumption to determine the actual cost of the contract. Then, it's important to add some extra percentage to account for the time between the calculation and the submission of the transaction.